pkg/operator/secretannotator/equinixmetal/reconciler_test.go

```diff
@@ -0,0 +1,198 @@
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+
+	configv1 "github.com/openshift/api/config/v1"
+	operatorv1 "github.com/openshift/api/operator/v1"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	minterv1 "github.com/openshift/cloud-credential-operator/pkg/apis/cloudcredential/v1"
+	"github.com/openshift/cloud-credential-operator/pkg/operator/constants"
+	schemeutils "github.com/openshift/cloud-credential-operator/pkg/util"
+)
+
+const (
+	testSecretName = "testsecret"
+	testNamespace  = "testproject"
+	testInfraName  = "testcluster-abc123"
+)
+
+func init() {
+	log.SetLevel(log.DebugLevel)
+}
+
+func TestSecretAnnotatorReconcile(t *testing.T) {
+	schemeutils.SetupScheme(scheme.Scheme)
+
+	tests := []struct {
+		name                    string
+		existing                []runtime.Object
+		expectErr               bool
+		validateAnnotationValue string
+	}{
+		{
+			name: "operator disabled",
+			existing: []runtime.Object{
+				testSecret(),
+				testOperatorConfigMap("true"),
+				testOperatorConfig(""),
+			},
+		},
+		{
+			name: "operator enabled",
+			existing: []runtime.Object{
+				testSecret(),
+				testOperatorConfigMap("false"),
+				testOperatorConfig(""),
+			},
+		},
+		{
+			name: "annotate passthrough mode",
+			// right now only passthrough mode is supported so any secret works
+			existing: []runtime.Object{
+				testSecret(),
+				testOperatorConfig(""),
+			},
+			validateAnnotationValue: constants.PassthroughAnnotation,
+		},
+		{
+			name:      "missing secret",
+			expectErr: true,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			mockCtrl := gomock.NewController(t)
+			defer mockCtrl.Finish()
+
+			infra := &configv1.Infrastructure{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "cluster",
+				},
+				Status: configv1.InfrastructureStatus{
+					Platform:           configv1.EquinixMetalPlatformType,
+					InfrastructureName: testInfraName,
+				},
+			}
+
+			existing := append(test.existing, infra)
+
+			fakeClient := fake.NewFakeClient(existing...)
+
+			rcc := &ReconcileCloudCredSecret{
+				Client: fakeClient,
+				Logger: log.WithField("controller", "testController"),
+			}
+
+			_, err := rcc.Reconcile(reconcile.Request{
+				NamespacedName: types.NamespacedName{
+					Name:      testSecretName,
+					Namespace: testNamespace,
+				},
+			})
+
+			if !test.expectErr && err != nil {
+				t.Fatalf("unexpected error: %v", err)
+			}
+
+			if test.validateAnnotationValue != "" {
+				validateSecretAnnotation(fakeClient, t, test.validateAnnotationValue)
+			}
+		})
+	}
+}
+
+func testSecret() *corev1.Secret {
+	s := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      testSecretName,
+			Namespace: testNamespace,
+		},
+		Data: map[string][]byte{
+			"username": []byte("someuser"),
```

Comment:
This seems decendent from:
 https://github.com/openshift/cloud-credential-operator/pull/283/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R137-R138 
while Equinix Metal uses:
https://github.com/openshift/cloud-credential-operator/pull/283/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R71
`EQUINIX_METAL_API_KEY`